pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -362,6 +371,12 @@ func (r *ReconcileClusterSync) Reconcile(request reconcile.Request) (reconcile.R
 
 	setFailedCondition(clusterSync)
 
+	// Set clusterSync.Status.FirstSyncSetsSuccessTime
+	syncStatuses := append(syncStatusesForSyncSets, syncStatusesForSelectorSyncSets...)
+	if len(syncStatuses) > 0 && clusterSync.Status.FirstSyncSetsSuccessTime == nil {
+		r.setFirstSyncSetsSuccessTime(syncStatuses, cd, clusterSync, logger)
+	}
```

Comment:
I might need some clarification here but this looks wrong. I would expect syncStatusesForSyncSets and syncStatusesForSelectorSyncSets contain statuses for failed attempts. We just check if the length > 0 but that looks true even if something failed. It feels like this if is probably meant to be:

```
if len(syncStatuses) > 0 && clusterSync.Status.FirstSyncSetsSuccessTime == nil && ! selectorSyncSetsNeedRequeue && !syncSetsNeedRequeue {
```

i.e. is nothing needs a requeue, we must have been successful.